Compilation error in ptrace() in odls_default_module.c on FreeBSD

Thank you for taking the time to submit an issue!

## Background information

Compilation error on FeeBSD 12 in ptrace() call

### What version of the PMIx Reference Server are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)

github master @ d64505a53fefa9f1952ae1ebf1880fb6ee723b81

##### What version of PMIx are you using? (e.g., v1.2.5, v2.0.3, v2.1.0, git branch name and hash, etc.)

github master @ a3cfa97da6983a33411e367f6a250964cff1dc55

### Please describe the system on which you are running

* Operating system/version: FreeBSD 12.1
* Computer hardware: x86_64
* Network type: N/A

-----------------------------

## Details of the problem

```
Making all in mca/odls/default
  CC       odls_default_module.lo
odls_default_module.c: In function 'do_parent':
odls_default_module.c:473:20: error: 'PTRACE_DETACH' undeclared (first use in this function); did you mean 'PRRTE_DETACH'?
  473 |             ptrace(PTRACE_DETACH, cd->child->pid, 0, (void*)SIGSTOP);
      |                    ^~~~~~~~~~~~~
      |                    PRRTE_DETACH
odls_default_module.c:473:20: note: each undeclared identifier is reported only once for each function it appears in
odls_default_module.c:473:54: warning: passing argument 4 of 'ptrace' makes integer from pointer without a cast [-Wint-conversion]
  473 |             ptrace(PTRACE_DETACH, cd->child->pid, 0, (void*)SIGSTOP);
      |                                                      ^
      |                                                      |
      |                                                      void *
In file included from odls_default_module.c:113:
/usr/include/sys/ptrace.h:220:57: note: expected 'int' but argument is of type 'void *'
  220 | int ptrace(int _request, pid_t _pid, caddr_t _addr, int _data);
      |                                                     ~~~~^~~~~
*** Error code 1
```

--- a/src/mca/odls/default/odls_default_module.c
+++ b/src/mca/odls/default/odls_default_module.c
@@ -470,7 +470,7 @@ static int do_parent(prrte_odls_spawn_caddy_t *cd, int read_fd)
                 return PRRTE_ERR_FAILED_TO_START;
             }
             errno = 0;
-            ptrace(PTRACE_DETACH, cd->child->pid, 0, (void*)SIGSTOP);
+            ptrace(PRRTE_DETACH, cd->child->pid, 0, (void*)SIGSTOP);
             if (0 != errno) {
                 /* couldn't detach */
                 cd->child->state = PRRTE_PROC_STATE_FAILED_TO_START;

